### PR TITLE
DNN-20025: Make PurgePage class testable

### PR DESCRIPTION
**Pull request by Mohtshum**
- Introduce a new class ContentVerifier : IContentVerifier, which takes IPortalGroupController and IPortalController as dependencies
- Add static ContentVerifier field in PortalHelper
- Make PortalHelper.IsContentExistsForRequestedPortal forward calls to ContentVerifier
- Mark PortalHelper obsolete

### CODE REVIEW

File: src/Dnn.PersonaBar.Library/Helper/PortalHelper.cs [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
-            return contentPortalId == portalSettings.PortalId 
-                || portalSettings == currentPortal 
-                || (checkForSiteGroup && IsRequestForSiteGroup(contentPortalId, portalSettings.PortalId));                
+        {        
+            return contentVerifier.IsContentExistsForRequestedPortal(contentPortalId, portalSettings, checkForSiteGroup);            
         }
 
-        private static bool IsRequestForSiteGroup(int portalId, int portalIdSiteGroup)
-        {
+        private bool IsRequestForSiteGroup(int portalId, int portalIdSiteGroup)
```
Comment (tpluscode) [inline src/Dnn.PersonaBar.Library/Helper/PortalHelper.cs]: This is no longer used and can be removed

File: src/Dnn.PersonaBar.Library/Helper/PortalHelper.cs [hunk 2]
```diff
@@ -1,20 +1,21 @@
 ﻿using DotNetNuke.Entities.Portals;
+using System;
 using System.Linq;
 
 namespace Dnn.PersonaBar.Library.Helper
 {
     public class PortalHelper
     {
+        static readonly IContentVerifier contentVerifier = new ContentVerifier(PortalController.Instance, PortalGroupController.Instance);
```
Comment (tpluscode) [inline src/Dnn.PersonaBar.Library/Helper/PortalHelper.cs]: Please add `private` modifier and rename in PascalCase

File: src/Dnn.PersonaBar.Library/Helper/ContentVerifier.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+using DotNetNuke.Entities.Portals;
+
+namespace Dnn.PersonaBar.Library.Helper
+{
+    public class ContentVerifier : IContentVerifier
+    {
+        private IPortalController _portalController;
+        private IPortalGroupController _portalGroupController;
+
+        public ContentVerifier(IPortalController portalController, IPortalGroupController portalGroupController)
```
Comment (tpluscode) [inline src/Dnn.PersonaBar.Library/Helper/ContentVerifier.cs]: Please add a parameterless constructor

**Review by tpluscode — CHANGES_REQUESTED** [review]
This is great. Just minor finishing touches and we'll merge ASAP

**Comment by Mohtshum** [comment]
Changes pushed

**Review by tpluscode — APPROVED** [review]

**Comment by tpluscode** [comment]
partially fixes dnnsoftware/Dnn.AdminExperience.Extensions#571

**Pull request merged**